e2e/src/usecases/WalletConnectV2.js

```diff
@@ -1,68 +1,164 @@
-import WalletConnectClient, { CLIENT_EVENTS } from '@walletconnect/client'
+import { newKit } from '@celo/contractkit'
+import Client from '@walletconnect/sign-client'
+import fetch from 'node-fetch'
 import { formatUri } from '../utils/encoding'
-import { scrollIntoView, sleep, waitForElementId } from '../utils/utils'
+import { enterPinUiIfNecessary, scrollIntoView, waitForElementId } from '../utils/utils'
 
-const WalletConnectAPIKey = process.env.WALLET_CONNECT_API_KEY || ''
-let uri, walletConnector
+const jestExpect = require('expect')
 
-export default WalletConnect = () => {
-  beforeAll(async () => {
-    await reloadReactNative()
-    // Create connector
-    walletConnector = await WalletConnectClient.init({
-      relayProvider: 'wss://relay.walletconnect.com',
-      apiKey: WalletConnectAPIKey,
-      metadata: {
-        name: 'WalletConnectV2 E2E',
-        description: 'WalletConnect Client',
-        url: 'https://valoraapp.com/',
-        icons: [],
+const dappName = 'WalletConnectV2 E2E'
+const DEFAULT_PROJECT_ID =
+  process.env.WALLET_CONNECT_PROJECT_ID || 'b8620cf33f99d407708c298d68224d8a'
+
+const kitUrl = process.env.FORNO_URL || 'https://alfajores-forno.celo-testnet.org'
+const kit = newKit(kitUrl)
+
+const walletAddress = (
+  process.env.E2E_WALLET_ADDRESS || '0x6131a6d616a4be3737b38988847270a64bc10caa'
+).toLowerCase()
+
+async function formatTestTransaction(address) {
+  try {
+    const response = await fetch(kitUrl, {
```

Comment:
i'm not sure if it's really necessary to get the real nonce here, but the example dapp that walletconnect provides does this. i think it will make the test less flakey if the nonce is correct (we're already using forno to check the tx receipt so it doesn't add an extra source of flake here)